app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarDetailsFragment.kt

```diff
@@ -129,17 +130,25 @@ class CalendarDetailsFragment : RoundedBottomSheetDialogFragment() {
     }
 
     private fun displayDeleteDialog(eventId: String) {
+        val s = TcaDb.getInstance(requireContext()).calendarDao().getSeriesIdForEvent(eventId)
         val alertDialog = AlertDialog.Builder(requireContext())
                 .setTitle(R.string.event_delete_title)
                 .setMessage(R.string.delete_event_info)
                 .setPositiveButton(R.string.delete) { _, _ -> deleteEvent(eventId) }
-                .setNegativeButton(R.string.cancel, null)
-                .create()
-
-        alertDialog.window?.setBackgroundDrawableResource(R.drawable.rounded_corners_background)
+                .setNeutralButton(R.string.cancel, null)
+        if (s!=null) {// a event series
+            alertDialog.setNegativeButton(R.string.delete_series) {_, _ ->deleteEventSeries(s)}
+        }
         alertDialog.show()
     }
 
+    private fun deleteEventSeries(seriesId: String){
+        val calendarItems = TcaDb.getInstance(requireContext()).calendarDao().getCalendarItemsInSeries(seriesId)
+        calendarItems.forEach{
+            deleteEvent(it.nr)
+        }
+    }
```

Comment:
This should probably be a function inside the calendarDao, so it can be potentially reused somewhere else.

Reply:
In the Dao we can only delete local entries, since we have to iterate over all events to delete them remotely I think this might be the better solution, what do you think?

Reply:
Sure, works either way. Probably won't need this a lot anywhere else.